pandas/core/indexes/base.py

```diff
@@ -4309,14 +4310,25 @@ def _concat(self, to_concat, name):
 
         if len(typs) == 1:
             return self._concat_same_dtype(to_concat, name=name)
-        return _concat._concat_index_asobject(to_concat, name=name)
+        return Index._concat_same_dtype(self, to_concat, name=name)
 
     def _concat_same_dtype(self, to_concat, name):
         """
         Concatenate to_concat which has the same class.
         """
         # must be overridden in specific classes
-        return _concat._concat_index_asobject(to_concat, name)
+        klasses = (ABCDatetimeIndex, ABCTimedeltaIndex, ABCPeriodIndex, ExtensionArray)
```

Comment:
should there be references to specific Index types in `pandas/core/indexes/base.py`?

Reply:
We've already got a few; in this case Index is serving as less of a base class and more of a ObjectIndex.